Sources/PackageCollections/PackageIndexAndCollections.swift

```diff
@@ -0,0 +1,284 @@
+    ) {
+        self.collections.refreshCollection(source, callback: callback)
+    }
+
+    public func addCollection(
+        _ source: PackageCollectionsModel.CollectionSource,
+        order: Int?,
+        trustConfirmationProvider: ((PackageCollectionsModel.Collection, @escaping (Bool) -> Void) -> Void)?,
+        callback: @escaping (Result<PackageCollectionsModel.Collection, Error>) -> Void
+    ) {
+        self.collections.addCollection(source, order: order, trustConfirmationProvider: trustConfirmationProvider, callback: callback)
+    }
+
+    public func removeCollection(
+        _ source: PackageCollectionsModel.CollectionSource,
+        callback: @escaping (Result<Void, Error>) -> Void
+    ) {
+        self.collections.removeCollection(source, callback: callback)
+    }
+
+    public func getCollection(
+        _ source: PackageCollectionsModel.CollectionSource,
+        callback: @escaping (Result<PackageCollectionsModel.Collection, Error>) -> Void
+    ) {
+        self.collections.getCollection(source, callback: callback)
+    }
+
+    public func listPackages(
+        collections: Set<PackageCollectionsModel.CollectionIdentifier>?,
+        callback: @escaping (Result<PackageCollectionsModel.PackageSearchResult, Error>) -> Void
+    ) {
+        self.collections.listPackages(collections: collections, callback: callback)
+    }
+
+    public func listTargets(
+        collections: Set<PackageCollectionsModel.CollectionIdentifier>?,
+        callback: @escaping (Result<PackageCollectionsModel.TargetListResult, Error>) -> Void
+    ) {
+        self.collections.listTargets(collections: collections, callback: callback)
+    }
+    
+    public func findTargets(
+        _ query: String,
+        searchType: PackageCollectionsModel.TargetSearchType?,
+        collections: Set<PackageCollectionsModel.CollectionIdentifier>?,
+        callback: @escaping (Result<PackageCollectionsModel.TargetSearchResult, Error>) -> Void
+    ) {
+        self.collections.findTargets(query, searchType: searchType, collections: collections, callback: callback)
+    }
+    
+    // MARK: - Package index APIs
+
+    public func isIndexEnabled(callback: @escaping (Result<Bool, Error>) -> Void) {
+        self.index.get { result in
+            switch result {
+            case .failure(PackageIndexError.featureDisabled):
+                callback(.success(false))
+            case .failure(let error):
+                callback(.failure(error))
+            case .success(let url):
+                callback(.success(url != .none))
+            }
+        }
+    }
+
+    public func setIndex(
+        url: Foundation.URL,
+        callback: @escaping (Result<Void, Error>) -> Void
+    ) {
+        self.index.set(url: url, callback: callback)
+    }
+
+    public func unsetIndex(callback: @escaping (Result<Void, Error>) -> Void) {
+        self.index.unset(callback: callback)
+    }
```

Comment:
similarly to the comment above the set/unset should not be part of this API imo, and should only be exposed on the storage data structure

Reply:
amended https://github.com/apple/swift-package-manager/pull/4010/commits/c1d364a022d867b5bfcd9f69006a1fc714001414